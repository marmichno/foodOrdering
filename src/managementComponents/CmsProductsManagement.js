import {CmsNavbar} from './CmsNavbar';

export const CmsProductsManagement = () => {
    const products = [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20];

    return(
        <div className="productManagementMainContainer">
            <CmsNavbar/>
            <div className="productManagementContent">
                <div className="productRow">
                    <div><h1>id</h1></div>
                    <div><h1>name</h1></div>
                    <div><h1>primary price</h1></div>
                    <div><h1>price including promotions</h1></div>
                    <div><h1>group</h1></div>
                    <div><h1>Edit</h1></div>
                </div>

                <div className="productRow">
                    <div><p>-</p></div>
                    <div className="addInput"><input type="text" placeholder="product name" autocomplete="off" name="category" required></input></div>
                    <div className="addInput"><input type="text" placeholder="product price" autocomplete="off" name="category" required></input></div>
                    <div><p>-</p></div>
                    <div className="addInput"><select><option>Sushi rolls</option><option>Ramen</option><option>Prawns</option><option>Others</option></select></div>
                    <div><button>save</button></div>
                </div>

                {products.map(value => {
                    return(
                        <div className="productRow">
                            <div><p>256</p></div>
                            <div><p>Philadelphia</p></div>
                            <div><p>6.00$</p></div>
                            <div><p>4.50$</p></div>
                            <div><p>Sushi rolls</p></div>
                            <div><button>Edit</button></div>
                        </div>
                    )
                })}
                
            </div>
        </div>
    )
}