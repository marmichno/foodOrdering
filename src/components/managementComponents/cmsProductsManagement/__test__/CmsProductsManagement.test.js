jest.unmock('axios');
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import {CmsProductsManagement} from '../CmsProductsManagement';
import "@testing-library/jest-dom/extend-expect";
import {render, fireEvent, waitFor, screen} from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux'; 
import mockedAxios from 'axios';
import validProductsJson from '../__mocks__/validProducts.json';
import validGroupsJson from '../__mocks__/validGroups.json';
import TestRenderer from "react-test-renderer";


const store = createStore(() => [], {}, applyMiddleware());
const mock = new MockAdapter(axios);

test("checks if products are visible after api call", async () => {

    mock.onGet("http://localhost:8080/api/product").reply(200, validProductsJson);
    mock.onGet("http://localhost:8080/api/group").reply(200, validGroupsJson);

    const {getAllByText} = render(<Provider store={store}><CmsProductsManagement/></Provider>);

    await waitFor(() => {
        expect(getAllByText('philadelphia'));
        expect(getAllByText('rice,nori,salmon,philadelphia cream'));
        expect(getAllByText('6.00$'));
    })
});

test("checks if products groups are visible after api call", async () => {

    mock.onGet("http://localhost:8080/api/product").reply(200, validProductsJson);
    mock.onGet("http://localhost:8080/api/group").reply(200, validGroupsJson);

    const {getAllByText, getByTestId} = render(<Provider store={store}><CmsProductsManagement/></Provider>);

    await waitFor(() => {
        fireEvent.click(getByTestId("groupSelect"));
    })

    await waitFor(() => {
        expect(getAllByText('sushi rolls'));
        expect(getAllByText('prawns'));
        expect(getAllByText('others'));
        expect(getAllByText('sets'));
    })
});

test("checks if product is visible/isnt visible after search", async () => {

    mock.onGet("http://localhost:8080/api/product").reply(200, validProductsJson);
    mock.onGet("http://localhost:8080/api/group").reply(200, validGroupsJson);

    const {getAllByText, findByTestId} = render(<Provider store={store}><CmsProductsManagement/></Provider>);

    const input = await findByTestId('searchbar');

    await waitFor(() =>{
        fireEvent.change(input, {target: { value:"philadelphia" }});
    })

    const prawns = screen.queryByText('prawn');

    await waitFor(() => {
        expect(getAllByText('philadelphia'));
        expect(getAllByText('rice,nori,salmon,philadelphia cream'));
        expect(getAllByText('6.00$'));
        expect(prawns).toBeNull();
    })
});

test("matches snapshot", () => {
    const tree = TestRenderer.create(<Provider store={store}><CmsProductsManagement/></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});
