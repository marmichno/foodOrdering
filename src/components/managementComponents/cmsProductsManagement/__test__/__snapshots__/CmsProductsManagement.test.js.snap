// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="productManagementMainContainer"
>
  <div
    className="hamburgerMenu"
  >
    <svg
      fill="none"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="3"
        x2="21"
        y1="12"
        y2="12"
      />
      <line
        x1="3"
        x2="21"
        y1="6"
        y2="6"
      />
      <line
        x1="3"
        x2="21"
        y1="18"
        y2="18"
      />
    </svg>
  </div>
  <div
    className="productManagementContent"
  >
    <div
      className="productRow"
    />
    <div
      className="productRow"
    >
      <div>
        <h1>
          id
        </h1>
      </div>
      <div>
        <h1>
          name
        </h1>
      </div>
      <div>
        <h1>
          price
        </h1>
      </div>
      <div>
        <h1>
          contains
        </h1>
      </div>
      <div>
        <h1>
          promotion
        </h1>
      </div>
      <div>
        <h1>
          group
        </h1>
      </div>
      <div>
        <h1>
          Edit
        </h1>
      </div>
    </div>
    <div
      className="productRow"
    >
      <div>
        <p>
          -
        </p>
      </div>
      <div
        className="addInput"
      >
        <input
          autoComplete="off"
          name="category"
          onChange={[Function]}
          placeholder="philadelphia"
          required={true}
          type="text"
        />
      </div>
      <div
        className="addInput"
      >
        <input
          autoComplete="off"
          name="category"
          onChange={[Function]}
          placeholder="6.00"
          required={true}
          type="text"
        />
      </div>
      <div
        className="addInput"
      >
        <input
          autoComplete="off"
          name="category"
          onChange={[Function]}
          placeholder="rice, nori, philadelphia cream"
          required={true}
          type="text"
        />
      </div>
      <div>
        <p>
          -
        </p>
      </div>
      <div
        className="addInput"
      >
        <select
          data-testid="groupSelect"
          onChange={[Function]}
        />
      </div>
      <div
        className="addButton"
        data-testid="addButton"
        onClick={[Function]}
      >
        <button>
          save
        </button>
      </div>
    </div>
    <div
      className="searchBar"
    >
      <input
        autoComplete="off"
        data-testid="searchbar"
        name="category"
        onChange={[Function]}
        placeholder="Search"
        required={true}
        type="text"
      />
      <h1>
        Search
      </h1>
    </div>
  </div>
</div>
`;
